master/buildbot/process/remotetransfer.py

```diff
@@ -128,7 +128,11 @@ def remote_unpack(self):
 
         # Unpack archive and clean up after self
         archive = tarfile.open(name=self.tarname, mode=mode)
-        archive.extractall(path=self.destroot)
+        try:
+            archive.extractall(path=self.destroot)
+        except OSError:
+            print("Error:{}is not a valid directory".format(self.destroot))
```

Comment:
print is going to be burried inside twisted.log.
We use them for debug only, so that we can grep for print, and quicly see leftover debug prints .
In that case, it would make more sense to display the error in a step's logs.
You can use
```suggestion
     self.step.addLogWithException("not a valid directory", e)
```